specs/ProcessInfo.md

```diff
@@ -0,0 +1,269 @@
+            var memoryInBytes = proc.PrivateMemorySize64;
+            var b2kb = memoryInBytes / 1024;
+            result = result + $"Process ID: {processId} | Process Kind: {kind} | Memory: {b2kb} KB\n";
+        }
+    }
+
+    MessageBox.Show(this, result, "Process List");
+}
+```
+    
+```cpp
+ProcessComponent::ProcessComponent(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    // Register a handler for the ProcessInfoChanged event.
+    //! [ProcessInfoChanged]
+    environment = appWindow->GetWebViewEnvironment();
+    CHECK_FAILURE(environment->add_ProcessInfoChanged(
+        Callback<ICoreWebView2ProcessInfoChangedEventHandler>(
+            [this,environment](ICoreWebView2Environment* sender, IUnknown* args) -> HRESULT {
+                CHECK_FAILURE(environment->get_ProcessInfo(&m_processCollection));
+
+                return S_OK;
+            })
+            .Get(),
+        &m_processInfoChangedToken));
+    //! [ProcessInfoChanged]
+}
+
+std::wstring ProcessComponent::ProcessKindToString(const COREWEBVIEW2_PROCESS_KIND kind)
+{
+    switch (kind)
+    {
+#define KIND_ENTRY(kindValue)                                                                  \
+    case kindValue:                                                                            \
+        return L#kindValue;
+
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_BROWSER);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_RENDERER);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_UTILITY);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_SANDBOX_HELPER);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_GPU);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_PPAPI_PLUGIN);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_PPAPI_BROKER);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_UNKNOWN);
+
+#undef KIND_ENTRY
+    }
+
+    return L"PROCESS KIND: " + std::to_wstring(static_cast<uint32_t>(kind));
+}
+
+// Get the process info
+//! [ProcessInfoChanged]
+void ProcessComponent::PerformanceInfo()
+{
+    std::wstring result;
+    UINT processListCount;
+    CHECK_FAILURE(m_processCollection->get_Count(&processListCount));
+
+    if (processListCount == 0)
+    {
+        result += L"No process found.";
+    }
+    else
+    {
+        result += std::to_wstring(processListCount) + L" process(s) found";
+        result += L"\n\n";
+        for (UINT i = 0; i < processListCount; ++i)
+        {
+            wil::com_ptr<ICoreWebView2StagingProcessInfo> processInfo;
+            CHECK_FAILURE(m_processCollection->GetValueAtIndex(i, &processInfo));
+
+            UINT32 processId = 0;
+            COREWEBVIEW2_PROCESS_KIND kind;
+            CHECK_FAILURE(processInfo->get_Id(&processId));
+            CHECK_FAILURE(processInfo->get_Kind(&kind));
+
+            WCHAR id[4096] = L"";
+            StringCchPrintf(id, ARRAYSIZE(id), L"Process ID: %u", processId);
+
+            HANDLE processHandle = OpenProcess(PROCESS_ALL_ACCESS, FALSE, processId);
```

Comment:
Handle is leaked. Save it in a `wil::unique_process_handle`.

Reply:
Would it be okay if I just `CloseHandle(processHandle)`?